src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -162,15 +182,20 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_ep_get_short,
                  uint64_t remote_addr, uct_rkey_t rkey)
 {
     uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_cuda_copy_iface_t);
-    uct_cuda_copy_stream_t idx   = UCT_CUDA_COPY_STREAM_D2H;
     ucs_status_t status;
+    ucs_memory_info_t src_info;
+    ucs_memory_info_t dst_info;
+    cudaStream_t *stream;
+
+    uct_cuda_copy_memory_detect(iface, (void *)remote_addr, (size_t)length,
+                                &src_info);
+    uct_cuda_copy_memory_detect(iface, buffer, (size_t)length, &dst_info);
 
-    UCT_CUDA_COPY_CHECK_AND_CREATE_STREAM(iface, idx);
+    stream  = uct_cuda_copy_get_stream(iface, src_info.type, dst_info.type);
```

Comment:
maybe define a wrapper for these 3 calls, as they seems to be frequently used together